File: src/0167.py
class Solution:
    def twoSum(self, numbers, target: int):
        imax = len(numbers) - 1
        imin = 0
        res = numbers[imin] + numbers[imax]
        while res != target:
            if res > target:
                imax -= 1
            else:
                imin += 1
            res = numbers[imin] + numbers[imax]
        return [imin + 1, imax + 1]

File: src/0167.cpp
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int l = 0, r = numbers.size() - 1;
        while(r > l) {
            int sum = numbers[l] + numbers[r];
            if(sum == target) {
                break;
            }else if(sum > target) {
                r--;
            }else {
                l++;
            }
        }
        return {l+1, r+1};
    }
};
